guides/choosing_mono_or_multi_repo.md

```diff
@@ -0,0 +1,96 @@
+# Choosing between a mono repo and a multi repo
+
+A mono repo is a single repository that contains all the code for multiple projects. A multi repo is a collection of repositories, each containing the code for a single project. Both approaches have their own advantages and disadvantages. This guide will help you decide which approach is best for your project.
+
+Developers should proactively choose between a mono repo and a multi repo based on the project requirements. The decision should be made at the beginning of the project, as it can be difficult to switch between the two later on.
+
+The decision should be reviewed periodically to ensure that it still meets the project requirements. If the project requirements change, it may be necessary to switch between a mono repo and a multi repo.
+
+## Considerations
+
+Teams should consider the following key drivers for choosing between a mono repo and a multi repo:
+- Size and complexity of the project
+- Technology choices
+- Collaboration
+- CI/CD
+- Release process
+- Commit history
+- Versioning
+- Security
+- Ownership
+
+### Size and complexity
+
+A mono repo is best suited for small, single language projects with a low number of dependencies. A multi repo is best suited for large, complex projects with multiple languages and dependencies.
+
+Larger projects can be difficult to manage in a mono repo, as the size and complexity of the codebase can make it difficult to navigate and understand. In this case, a multi repo may be a better choice, as it allows teams to work on smaller, more manageable codebases.
+
+Size of downloading the repository is also a consideration. A mono repo can be large and take a long time to download, particularly for new team members. A multi repo can be smaller and quicker to download, as each repository contains a smaller amount of code.
+
+Performance of scanning tools such as secret detection tools is also increased with smaller repositories.
+
+### Technology choices
+
+Mono repos are best suited for projects that use a single technology stack. Multi repos are best suited for projects that use multiple technology stacks.
```

Comment:
Is the monorepo tooling also pretty much specific to a single technology?

Reply:
Yep, good point.  Added a reference to this.